test/layout_test.exs

```diff
@@ -0,0 +1,118 @@
+defmodule LayoutTest do
+  use ExUnit.Case
+
+  alias Layout.{Key, LED}
+
+  # The following tests use parameters defined in `config/host/test.exs`
+  #
+  # The following keys are defined for the test environment:
+  # keys: [
+  #   %{id: :k1, x: 0, y: 0, opts: [led: :l1]},
+  #   %{id: :k2, x: 2, y: 1.5, opts: [width: 1.5, height: 2, led: :l2]},
+  #   %{id: :k3, x: 5, y: 0}
+  # ]
+  #
+  # The following LEDs are defined for the test environment:
+  # leds: [
+  #   %{id: :l1, x: 0, y: 0},
+  #   %{id: :l2, x: 2, y: 1.5},
+  #   %{id: :l3, x: 3, y: 3}
+  # ]
+  test "Layout.load_from_config/0 loads the layout defined in the application config" do
+    assert Layout.load_from_config() == %Layout{
+             keys: [
+               %Key{height: 1, id: :k1, led: :l1, width: 1, x: 0, y: 0},
+               %Key{height: 2, id: :k2, led: :l2, width: 1.5, x: 2, y: 1.5},
+               %Key{height: 1, id: :k3, led: nil, width: 1, x: 5, y: 0}
+             ],
+             keys_by_leds: %{
+               l1: %Key{height: 1, id: :k1, led: :l1, width: 1, x: 0, y: 0},
+               l2: %Key{height: 2, id: :k2, led: :l2, width: 1.5, x: 2, y: 1.5}
+             },
+             leds: [
+               %LED{id: :l1, x: 0, y: 0},
+               %LED{id: :l2, x: 2, y: 1.5},
+               %LED{id: :l3, x: 3, y: 3}
+             ],
+             leds_by_keys: %{
+               k1: %LED{id: :l1, x: 0, y: 0},
+               k2: %LED{id: :l2, x: 2, y: 1.5}
+             }
+           }
+  end
+
+  defp add_layout(_context) do
+    [layout: Layout.load_from_config()]
```

Comment:
Rather than loading the layout from the config to use for unit tests, we may want to use a test fixture.